south-african-theological-seminary.csl

```diff
@@ -0,0 +1,263 @@
+      <label form="short" prefix=" (" suffix=")"/>
+      <substitute>
+        <names variable="editor"/>
+        <names variable="translator"/>
+        <text macro="title" suffix="."/>
+        <text variable="publisher"/>
+      </substitute>
+    </names>
+  </macro>
+  <macro name="author-short">
+    <names variable="author">
+      <name form="short" and="text" initialize-with=""/>
+      <substitute>
+        <names variable="editor"/>
+        <names variable="translator"/>
+        <choose>
+          <if type="bill book graphic legal_case motion_picture report song" match="any">
+            <text variable="title" form="short"/>
+          </if>
+          <else>
+            <text variable="title" form="short" quotes="true"/>
+          </else>
+        </choose>
+        <text variable="publisher"/>
+      </substitute>
+    </names>
+  </macro>
+  <macro name="author-count">
+    <names variable="author">
+      <name form="count"/>
+      <substitute>
+        <names variable="editor"/>
+        <names variable="translator"/>
+        <choose>
+          <if type="bill book graphic legal_case motion_picture report song" match="any">
+            <text variable="title" form="short"/>
+          </if>
+          <else>
+            <text variable="title" form="short" quotes="true"/>
+          </else>
+        </choose>
+      </substitute>
+    </names>
+  </macro>
+  <macro name="title">
+    <choose>
+      <if type="bill book graphic legal_case motion_picture song" match="any">
+        <text variable="title" font-style="italic"/>
+      </if>
+      <else>
+        <text variable="title"/>
+      </else>
+    </choose>
+  </macro>
+  <macro name="issued">
+    <date variable="issued">
+      <date-part name="year"/>
+    </date>
+  </macro>
+  <macro name="publisher">
+    <group delimiter=": ">
+      <text variable="publisher-place"/>
+      <text variable="publisher"/>
+    </group>
+  </macro>
+  <macro name="edition">
+    <group delimiter=" " prefix="(" suffix=")">
+      <number variable="edition" form="ordinal"/>
+      <label variable="edition"/>
+    </group>
+  </macro>
+  <macro name="volume">
+    <group delimiter=" " prefix="(" suffix=")">
+      <number variable="volume"/>
+      <label variable="volume" form="short"/>
+    </group>
+  </macro>
+  <macro name="date">
+    <choose>
+      <if match="any" is-uncertain-date="issued">
+        <text term="no date" form="short"/>
+      </if>
+      <else>
+        <choose>
+          <if type="article-newspaper" match="any">
+            <date variable="issued">
+              <date-part name="year"/>
+              <date-part name="day" prefix=" (" suffix=" "/>
+              <date-part name="month" suffix=")"/>
+            </date>
+          </if>
+          <else>
+            <text macro="issued"/>
+          </else>
+        </choose>
+      </else>
+    </choose>
+  </macro>
```

Comment:
@adam3smith 
Can you advise on the date macro?
Something funky going on for me.
Ideally it should give year, month, date for article-newspaper, otherwise for everything else just the year. if not date replace with "n.d.". Something fishy going on for me in the editor. With this code I get the dates, but no n.d....

Reply:
@POBrien333, the test is wrong.

```xml
      <if match="any" is-uncertain-date="issued">
        <text term="no date" form="short"/>
      </if>
```

should be

```xml
      <if variable="issued" match="none">
        <text term="no date" form="short"/>
      </if>
```

(see also http://docs.citationstyles.org/en/1.0.1/specification.html#choose and http://docs.citationstyles.org/en/1.0.1/specification.html#approximate-dates)

Reply:
That said, the logic might be a bit clearer if the standard case would assume the presence of an issued date, i.e.:

```xml
      <if variable="issued">
        ...
      </if>
      <else>
        <text term="no date" form="short"/>
      </else>
```

Reply:
yeah. I see now I need to use the variable and not this is-uncertain-date thing. thanks :)